### PR TITLE
feat: update ui to handle more nav items

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

To accommodate more nav items, we needed to re-think the structure of our menus. While doing this, we eliminated some things that were good ideas when we first started this project but ended up being unnecessary or counter intuitive like:
- Routes only have one single component in the config
- The AssetHeader component was broken up into individual components vs one larger component
- The back button being replaced with a much more useful breadcrumb component

This will allow us to scale the UI more easily, and have a very structure site architecture that we can lean on when building out components. Trimming down the route config also helps us to more easily upgrade to reacr-router v6 -- they only support one single component per route, in addition to a layout component.

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [ ] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Risk

This is a re-work of the current UI, it is mostly refactoring of existing code into smaller chunks/components. Its medium risk in that we moved some stuff around but overall didn't modify too much of the existing code. Although there may be side-effects of this, possibly performance losses or gains.

## Testing

For testing, use the platform as normal. Make sure all normal functions are accessible and there are no visual issues or performance issues.

## Screenshots (if applicable)
Screen Shot 2022-02-23 at 12 31 51 AM
Screen Shot 2022-02-23 at 12 32 00 AM
Screen Shot 2022-02-23 at 12 32 23 AM

### CODE REVIEW

**Comment by cjthompson** [comment]
I realize this is a draft, but I think it would be nice to have
1. screenshots of the changes with some description
2. a link to the wireframes/designs or product ticket

There should be some way for people to participate in a conversation about the changes (questions, comments, etc).

**Comment by cjthompson** [comment]
Can you please fill out the **Issue**, **Risk**, and **Testing** sections of the PR description?

File: src/pages/Accounts/AccountToken/AccountToken.tsx [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-import { Page } from 'components/Layout/Page'
 import { AccountSpecifier } from 'state/slices/portfolioSlice/portfolioSlice'
 import { selectPortfolioAccountIdsSortedFiat } from 'state/slices/selectors'
 export type MatchParams = {
   accountId: AccountSpecifier
   assetId: CAIP19
 }
 
-export const AccountToken = () => {
+export const AccountToken = ({ route }: { route: Route }) => {
```
Comment (gomesalexandre) [inline src/pages/Accounts/AccountToken/AccountToken.tsx:13]: Non-blocking, it's usually nice-to-have props as their own Type, either as an explicit props typing or a generic of `React.FC<T>`

**Review by gomesalexandre — APPROVED** [review]
This looks good to me from a code perspective. Great componentization, and great additions.
During full regression testing, I also found nothing major breaking on prod. features, no warning, no errors.
I've noticed however that:

1. plugins are breaking and both cosmos and osmosis routes are wrong, producing an infinite skeleton state:

- Osmosis route routes to `/assets/cosmos/:osmosis-1/:assetSubId`
- Cosmos route routes to `assets/cosmos/:cosmoshub-4/:assetSubId`

image

This is only affecting clicking on the plugin icons. Clicking Cosmos from assets routes properly. FYI, currently on `origin/develop`, enabling the investor flags do not render these icons anymore.

2. The search autocomplete shows native HTML select element on top of the styled one:

image

3. `/defi/earn` takes a lot of time to initially load and the main thread is blocked (cursor stays in pointer mode and hovering anywhere on the app doesn't give any hover feedback). This is not specific to this PR and is also happening in latest develop, but I've noticed the main thread is blocked for longer here (going from `/defi` to `/defi/earn` takes ~3 seconds on develop vs. ~13 seconds on this branch)

I will not block the PR for these, as it can be done in follow-up PR(s) to not make this one lengthier.

File: src/pages/Defi/views/Overview.tsx [hunk 2]
```diff
@@ -1,17 +1,26 @@
...
 
 import { OverviewHeader } from '../components/OverviewHeader'
 import { VaultList } from '../components/VaultList'
 import { useEarnBalances } from '../hooks/useEarnBalances'
 
-export const Overview = () => {
+const DefiHeader = () => {
+  return (
+    <Box>
+      <Heading>Defi</Heading>
```
Comment (0xdef1cafe) [inline src/pages/Defi/views/Overview.tsx:14]: duplicated and needs translations

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**